File: projects/m1/029-decimal-to-binary/solutions/enricozoboli/solution_decimal_to_binary.py
while True:
    number = int(input("Enter a decimal number: "))
    q = number
    result = ""
    
    if number == 0:
        print("The binary number of 0 is 0")
        break
    else:
        while q > 0 :
            r = q % 2
            result += str(r)
            q //= 2
        break

print(f"The binary number of {number} is {result[::-1]}")

File: projects/m1/029-decimal-to-binary/solutions/valentinaBazzacchi/solution-decimal-to-binary.js
let num;
let rest;
let result = "";
let onlyNumbers = /^[0-9]+$/;

num = prompt("Enter an integer to convert in binary");

if (!onlyNumbers.test(num) || num == 0){
alert("Not valid entry!");
} else {
    while (num > 0) {    
        rest = num % 2;
        result += String(rest);
        num = Math.floor(num / 2);
    } // fine while
alert("The binary is \n" + result);
} // fine else
  
//------------------------------------------------------//

// Method to convert a decimal number in binary
let num = +prompt("Enter an integer to convert in binary");
alert( "The binary is \n" + num.toString(2) );